cdk/lib/mparticle-api.ts

```diff
@@ -98,16 +117,15 @@ export class MParticleApi extends GuStack {
 			}),
 		);
 
-		// Allow the lambda to assume the roles that allow cross-account fetching of tags
-		// lambda.addToRolePolicy(
-		// 	new PolicyStatement({
-		// 		actions: ['sts:AssumeRole'],
-		// 		effect: Effect.ALLOW,
-		// 		resources: [
-		// 			// mobileAccountRoleArn.valueAsString,
-		// 			// targetingAccountRoleArn.valueAsString,
-		// 		],
-		// 	}),
-		// );
+		new SrLambdaDomain(this, {
+			subdomain: 'mparticle-api',
+			restApi: apiGateway.api,
+		});
+
+		new CfnOutput(this, 'MParticleLambdaRoleArn', {
+			value: mparticleLambdaRole.roleArn,
+			description: 'ARN of the mParticle Lambda execution role',
+			exportName: `${app}-${this.stage}-lambda-role-arn`,
```

Comment:
unless the exports are global you can export it without the extra info
```suggestion
			exportName: `lambda-role-arn`,
```

Reply:
@johnduffell I could be mistaken but I believe that there is a problem with your suggestion of using a generic export name. CloudFormation export names must be globally unique within an AWS region. If we use just lambda-role-arn, we'll run into issues:
- Naming conflicts: If we have multiple stacks or deploy to multiple stages, they'll all try to create the same export name
- Deployment failures: CloudFormation will fail with "Export already exists" errors
- Cross-stage conflicts: Your PROD and CODE stages would conflict with each other

Should we still try this?

Reply:
ok that fails my condition "unless the exports are global" which I did not check.